### PR TITLE
Closes #6604 Add fetch priority to LCP when url is relative

### LINKED ISSUE #70: Minify adds path to css rules with url() even if not needed
know that we don't handle the minification library, but I report the issue anyway.
With css and svg, you can use the mask rule with an url like this : url(#id)
the minification process seems to add the path to the theme folder even if not needed here, since it's not a file that is requested.

### LINKED ISSUE #6604: Fetchpriority of LCP is not applied on the LCP markup when the LCP URL is relative
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version: Commit 1b6f1972a78649e317463b5c5eb8b2e3574052b5
- Used the search feature to ensure that the bug hasn’t been reported before ✅ 

**Describe the bug**
The LCP feature should add the fetchpriority attribute to the LCP markup.
When the LCP markup contains a relative URL, this is not done. Note that the preload markup is correctly added with an absolute URL and the fetchpriority.

**To Reproduce**
1. Make sure to have LCP feature enabled (by default, it is). 
2. Clear the cache and critical images.
3. Manually go to xxx.e2e.rocketlabsqa.ovh/lcp_regular_image_template/ in incognito. Make sure that the LCP/ATF table is filled with information from this page in the DB. The LCP image should be `wp-content/rocket-test-data/images/600px-Mapang-test.gif`
4. Purge this URL
5. Re-load the page.
6. Check in the dev tools the markup of `wp-content/rocket-test-data/images/600px-Mapang-test.gif` image. There is no fetchpriority attribute.

If you do the same on http://xxx.e2e.rocketlabsqa.ovh/landing-page/, the attribute is correctly applied.

I think the difference is the absolute/relative paths.

Note that we have similar issues when a CDN is being used: if the URL in the DB contains the CDN hostname, then the fetchpriority is not applied as expected when serving the page.

**Expected behavior**
- Fetchpriority attribute must be set to the LCP markup when the LCP markup contains an absolute URL ✅ 
- Fetchpriority attribute must be set to the LCP markup when the LCP markup contains a relative URL 🔴 
- Built-in tests should cover those cases.
If it make sense to tackle this in the same issue (nice to have, otherwise, create a dedicated GH issue):
- Fetchpriority attribute must be set to the LCP markup when the a CDN is used and the URL stored in the LCP/ATF table as LCP contains the CDN URL.

**Additional context**
The issue seems to be with `set_fetchpriority` which performs a regex based on the URL we have in the DB. Mismatch between relative/absolute/CDN hostname make the regex fail?

**Acceptance Criteria**
- Repeat the How to reproduce on the two templates: /lcp_regular_image_template/  & /landing-page/. In both cases, the fetchpriority attribute must be set properly to the LCP markup.
- NRT with Rocket-E2E
- Tests (unit or integration) covers the relative path case and the absolute path case.
If dealt with in this PR, the same ACs must be checked with the CDN feature enabled and LCP stored in DB with the CDN hostname.

### PR DESCRIPTION
**Pull request by Khadreal**
# Description

Fixes #6604 

## Documentation
Modify the regex pattern, to capture all possible cases, absolute, relative and other valid urls 
### User documentation

*Explain how this code impacts users.*

### Technical documentation

*Explain how this code works. Diagram & drawings are welcomed.*

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue).

## New dependencies
N/A

## Risks
*List possible performance & security issues or risks, and explain how they have been mitigated.*

# Checklists

## Feature validation
- [ ] I validated all the Acceptance Criteria. If possible, provide sreenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [x] I implemented built-in tests to cover the new/changed code.

## Code style
- [x] I wrote self-explanatory code about what it does.
- [x] I wrote comments to explain why it does it.
- [x] I named variables and functions explicitely.
- [x] I protected entry points against unexpected inputs.
- [x] I did not introduce unecessary complexity.
- [x] I listed the introduced external dependencies explicitely on the PR.
- [x] I validated the repo-specific guidelines from CONTRIBUTING.md.

## Observability
- [x] I handled errors when needed.

### CODE REVIEW

File: inc/Engine/Media/AboveTheFold/Frontend/Controller.php [hunk 1]
```diff
@@ -146,9 +146,17 @@ private function set_fetchpriority( $lcp, string $html ): string {
 			return $html;
 		}
 
-		$url  = preg_quote( $lcp->src, '/' );
+		$url     = preg_quote( $lcp->src, '/' );
+		$pattern = '#<img(?:[^>]*?\s+)?src=["\']' . $url . '["\'](?:\s+[^>]*?)?>#';
+		if ( wp_http_validate_url( $lcp->src ) && ! $this->is_external_file( $lcp->src ) ) {
+			$url = get_rocket_parse_url( $lcp->src );
```
Comment (wordpressfan) [inline inc/Engine/Media/AboveTheFold/Frontend/Controller.php]: So here we only need the path? 

```suggestion
			$path = wp_parse_url( $lcp->src, PHP_URL_PATH );
```

File: inc/Engine/Media/AboveTheFold/Frontend/Controller.php [hunk 2]
```diff
@@ -146,9 +146,17 @@ private function set_fetchpriority( $lcp, string $html ): string {
...
+			$url = preg_quote( $url['path'], '/' );
```
Comment (wordpressfan) [inline inc/Engine/Media/AboveTheFold/Frontend/Controller.php]: why do we need to use preg_quote again? we already used it above with the whole url and I believe this is enough, what do u think?

**Review by wordpressfan — CHANGES_REQUESTED** [review]
Good job, just small enhancements

File: inc/Engine/Media/AboveTheFold/Frontend/Controller.php (hunk 2 above)
Comment (Khadreal) [reply inc/Engine/Media/AboveTheFold/Frontend/Controller.php]: The `$url` above is overwritten, preg_quote is use here for the path

**Review by wordpressfan — APPROVED** [review]
Good job

**Comment by Mai-Saad** [comment]
@Khadreal thanks for the PR. Working for the main shared template, and not adding fetchpriority in some edge cases as per exploratory testing below (not working with absolute URL too and same on trunk):

1. If the relative URL has fetchpriority='high' in the URL  ` `
2. If src was capital or has 1st letter capital or has space after it (If instead of `src="imageURL"` , `SRC` or `Src` or `src = "imageURL"`)
3. If a commented exact markup was 1st on page, fetchpriority will be added to the commented markup instead of active one 
```
<!-- <img src="/wp-content/rocket-test-data/images/lcp/testavif.avif"> -->

		<img src="/wp-content/rocket-test-data/images/lcp/testavif.avif">
```
4. LCP Image with Fetchpriority in markup, will have fetchpriority='high' and that will cause markup error. Example `` => seems regression and not on trunk @Khadreal can you please check?

@piotrbak Do you think we need GH for any of those cases 1-3?

**Comment by piotrbak** [comment]
@Mai-Saad I think those are edge cases, we should be good here

**Comment by Mai-Saad** [comment]
Related test plan https://wpmediaqa.testrail.io/index.php?/runs/view/854&group_by=cases:section_id&group_order=asc

**Comment by Khadreal** [comment]
@Mai-Saad handle the regression and 1 and 2 edge cases. We keep looking on how to resolve the 3rd one. You can do another test when you have some time.

**Comment by Khadreal** [comment]
@Mai-Saad I have added a fix for scenario 3 when there's a commented exact markup.

**Review by Mai-Saad — APPROVED** [review]
Working fine now
[testrail-report-606.pdf](https://github.com/wp-media/wp-rocket/files/15455222/testrail-report-606.pdf)
